test/end2end_test.go

```diff
@@ -85,29 +85,29 @@ func (s *testServer) EmptyCall(ctx context.Context, in *testpb.Empty) (*testpb.E
 	return new(testpb.Empty), nil
 }
 
-func newPayload(t testpb.PayloadType, size int32) *testpb.Payload {
+func newPayload(t testpb.PayloadType, size int32) (*testpb.Payload, error) {
 	if size < 0 {
-		grpclog.Fatalf("Requested a response with invalid length %d", size)
+		return nil, fmt.Errorf("Requested a response with invalid length %d", size)
 	}
 	body := make([]byte, size)
 	switch t {
 	case testpb.PayloadType_COMPRESSABLE:
 	case testpb.PayloadType_UNCOMPRESSABLE:
-		grpclog.Fatalf("PayloadType UNCOMPRESSABLE is not supported")
+		return nil, errors.New("PayloadType UNCOMPRESSABLE is not supported")
```

Comment:
Done.

On Fri, Sep 11, 2015 at 2:23 PM, Qi Zhao <notifications@github.com> wrote:

> In test/end2end_test.go
> <https://github.com/grpc/grpc-go/pull/333#discussion_r39301088>:
>
> >  	}
> >  	body := make([]byte, size)
> >  	switch t {
> >  	case testpb.PayloadType_COMPRESSABLE:
> >  	case testpb.PayloadType_UNCOMPRESSABLE:
> > -		grpclog.Fatalf("PayloadType UNCOMPRESSABLE is not supported")
> > +		return nil, errors.New("PayloadType UNCOMPRESSABLE is not supported")
>
> Please make them consistent. I think the fundamental reason to use
> errors.New instead of fmt.Errorf is when your code and/or tests need to
> reuse this error in multiple places.
>
> Whether it can Interpolate arguments is just a artifact in my mind. You
> can always use fmt.Sprintf as input param of errors.New.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-go/pull/333/files#r39301088>.
>